client/jetpack-connect/index.js

```diff
@@ -7,10 +7,15 @@ import page from 'page';
 /**
  * Internal dependencies
  */
+import userFactory from 'lib/user';
 import * as controller from './controller';
+import { login } from 'lib/paths';
 import { siteSelection } from 'my-sites/controller';
 
 export default function() {
+	const user = userFactory();
+	const isLoggedOut = ! user.get();
+
```

Comment:
Is it possible to use redux like we do here:

https://github.com/Automattic/wp-calypso/blob/db7b56c07cea7ecdb94ea4fdc815737c5c42298e/client/jetpack-connect/authorize-form.jsx#L101

Reply:
Yeah, good call, I'll change it.

Reply:
@sirreal is there a way to get hold of `state` here?

Reply:
😬  Probably not here in the routes. I think we'd need to go into a route handler in the controller, then we'll have access to the redux store.

Reply:
May as well leave as-is in that case.

Reply:
It could be a middleware route handler on the plans route

Reply:
We don't even need to go to the controller, we just need to get access to the context so this needs to happen inside a handler.

On that note, conditional route definitions depending on application state seem like something to avoid. I don't believe routes will be recalculated as state changes. What do you think?

Reply:
These aren't conditional route definitions, this code is executed for every request. Works fine elsewhere: https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/themes/index.web.js#L23

Reply:
> These aren't conditional route definitions, this code is executed for every request

No, I must be wrong here ^

Reply:
Run on boot or chunk load, so log out / log in should always force re-execution of this code. I prefer the more declarative style of having separate definitions here in `index.js`, especially in the theme example above where there is a whole different set of routes for logged-out.

Reply:
Sounds good, I'll give this a spin 👌